bitnami/kubeapps/values.yaml

```
@@ -1656,7 +1656,7 @@ kubeappsapis:
           ## @param kubeappsapis.pluginConfig.kappController.packages.v1alpha1.defaultUpgradePolicy Default upgrade policy generating version constraints
           ## enum: [ "major", "minor", "patch", "none" ]
           defaultUpgradePolicy: none
-          ## @param kubeappsapis.pluginConfig.kappController.packages.v1alpha1.defaultPrereleasesVersionSelection Default policy for allowing prereleases containing one of the identifiers
+          ## @param kubeappsapis.pluginConfig.kappController.packages.v1alpha1.defaultPrereleasesVersionSelection [nullable] Default policy for allowing prereleases containing one of the identifiers
```

If this field should be a string, you would need to add:
```suggestion
          ## @param kubeappsapis.pluginConfig.kappController.packages.v1alpha1.defaultPrereleasesVersionSelection [string,nullable] Default policy for allowing prereleases containing one of the identifiers
```

in order to show type string into the schema

---

This param can be:

````yaml
defaultPrereleasesVersionSelection: null
defaultPrereleasesVersionSelection: []
defaultPrereleasesVersionSelection: ["foo", "bar"]
````

So, I'm not sure we would want to show `type: string` in the schema, perhaps

```suggestion
          ## @param kubeappsapis.pluginConfig.kappController.packages.v1alpha1.defaultPrereleasesVersionSelection [array, nullable] Default policy for allowing prereleases containing one of the identifiers
```

---

sure, then you should add `array` where I put `string`. You assumption is correct, just do not add spaces (I am not sure right now if there is a trim into the code for that

---

I've added it, but it results in an invalid openapi schema: the required `items` property for arrays is not being generated.

#### With `@param .... [nullable]`, valid schema, but not representing the reality

````jsonc
"defaultPrereleasesVersionSelection": {
    "type": "object", 
    "description": "Default policy for allowing prereleases containing one of the identifiers",
    "default": "null",
    "nullable": true
},
````

#### With `@param .... [array, nullable]`, invalid schema

````jsonc
"defaultPrereleasesVersionSelection": {
    "type": "array", 
     // missing 'items' property
    "description": "Default policy for allowing prereleases containing one of the identifiers",
    "default": "null",
    "nullable": true
},
````

#### Expected valid OpenAPI 3.0.X schema:

_Note that we are explicitly not targeting openapi 3.1, otherwise, the `nullable` prop would be removed, and the `type` object would become an array. [Useful SO post clarifying the differences](https://stackoverflow.com/questions/48111459/how-to-define-a-property-that-can-be-string-or-null-in-openapi-swagger)._

````jsonc
"defaultPrereleasesVersionSelection": {
    "type": "string",
    "items": {
      "type": "string" // how can we infer this value? 
    },
    "description": "Default policy for allowing prereleases containing one of the identifiers",
    "default": null,
    "nullable": true
  }
````

---

So, rather than generating an invalid schema, I'd go with leaving it as is, until the `items` can be added/inferred in this edge case. WDYT?

---

Looks like another corner case for arrays. Thank you for sharing it.
> >So, rather than generating an invalid schema, I'd go with leaving it as is,

Currently, if you don't specify any modifier for setting the type, it will be object by default. I think if you add just `[nullable]` the `type` will be set to `object` into the schema. Is that what you mean?

---

Yes, as we are not using the schema currently, semantic correctness is not a priority, so having `type: object` is something bearable until this edge case gets sorted out.
I mean, sacrificing semantic over syntactic correctness is preferred in this case.

Do you want me to file an issue in the readme's repo? Can we merge this PR then?

---

As discussed offline, this corner case is going to be resolved soon in the readme-generator tool, therefore, let me update the PR to also include the [array] modifier.
Thanks!

---

I agree, let's proceed with this PR, I will try to fix t